pkg/types/vsphere/validation/platform.go

```diff
@@ -73,53 +73,53 @@ func validateMultiZone(p *vsphere.Platform, fldPath *field.Path) field.ErrorList
 			Password:    p.Password,
 			Datacenters: []string{p.Datacenter},
 		})
+	}
 
-		// populate failure domains that dont explicitly define a server
-		for idx, failureDomain := range p.FailureDomains {
-			if len(failureDomain.Server) == 0 {
-				p.FailureDomains[idx].Server = p.VCenter
-			}
-			if len(failureDomain.Topology.Datacenter) == 0 {
-				p.FailureDomains[idx].Topology.Datacenter = p.Datacenter
-			}
-			if len(failureDomain.Topology.ComputeCluster) == 0 {
-				p.FailureDomains[idx].Topology.ComputeCluster = fmt.Sprintf("/%s/host/%s", p.Datacenter, p.Cluster)
-			}
-			if len(failureDomain.Topology.Networks) == 0 && len(p.Network) > 0 {
-				if len(failureDomain.Topology.Networks) == 0 {
-					p.FailureDomains[idx].Topology.Networks = []string{p.Network}
-				}
-			}
-			if len(failureDomain.Topology.Datastore) == 0 {
-				p.FailureDomains[idx].Topology.Datastore = p.DefaultDatastore
+	// populate failure domains that dont explicitly define a server
+	for idx, failureDomain := range p.FailureDomains {
```

Comment:
This is now run all of the time (not just when `vcenters` is empty). This changes makes me a bit nervous due to how sensitive setting defaults has been.

Reply:
Yeah, unit tests failed everywhere. Still looking at the best way to solve setting default resource pool.